test/unit/org/apache/cassandra/utils/Generators.java

```diff
@@ -492,13 +494,20 @@ public static <T> Gen<Set<T>> set(Gen<T> gen, Gen<Integer> sizeGen)
         };
     }
 
-    public static <T extends Comparable<? super T>> Gen<List<T>> uniqueList(Gen<T> gen, Gen<Integer> sizeGen)
+    public static <T> Gen<List<T>> uniqueList(Gen<T> gen, Gen<Integer> sizeGen)
     {
-        return set(gen, sizeGen).map(t -> {
-            List<T> list = new ArrayList<>(t);
-            list.sort(Comparator.naturalOrder());
-            return list;
-        });
+        return rnd -> {
+            int size = sizeGen.generate(rnd);
+            Set<T> set = Sets.newHashSetWithExpectedSize(size);
+            List<T> output = new ArrayList<>(size);
+            for (int i = 0; i < size; i++)
+            {
+                T value;
+                while (!set.add(value = gen.generate(rnd))) {}
```

Comment:
this isn't as safe as `uniqueBestEffort` from accord... once accord merges we can talk about unifying all this stuff...

Reply:
for now no.  There are better impl in `Gens` and when i migrate off of quick theories we can just use the better versions; so no reason to handle here...